runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -151,13 +151,27 @@ void J9::RecognizedCallTransformer::process_java_lang_StringCoding_encodeASCII(T
    TR::SymbolReference *methodSymRef = comp()->getSymRefTab()->findOrCreateEncodeASCIISymbol();
    TR::Node *encodeASCIINode = TR::Node::createWithSymRef(TR::call, 3, methodSymRef);
 
-   TR::Node *newInputNode = TR::Node::create(sourceArrayNode, comp()->target().is64Bit() ? TR::aladd : TR::aiadd, 2);
-   TR::Node *arrayHeaderSizeNode = TR::Node::create(sourceArrayNode, comp()->target().is64Bit() ? TR::lconst : TR::iconst, 0, (int32_t)TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
+   TR::Node *newInputNode = NULL;
+   TR::Node *arrayHeaderSizeNode = NULL;
+   TR::Node *newOutputNode = NULL;
+
+   if (comp()->target().is64Bit())
+      {
+      newInputNode = TR::Node::create(sourceArrayNode, TR::aladd, 2);
+      newOutputNode = TR::Node::create(destinationArrayNode, TR::aladd, 2);
+      arrayHeaderSizeNode = TR::Node::create(sourceArrayNode, TR::lconst, 0, (int64_t)TR::Compiler->om.contiguousArrayHeaderSizeInBytes());
```

Comment:
Maybe it is better to use the `lconst` routine here https://github.com/eclipse/omr/blob/385cb28808891d8afb266c0a17a60e0f37bf9d86/compiler/il/OMRNode.cpp#L1399

Reply:
I'm now using `lconst` here and `iconst` in the else block below. Change is in: 0e36ab6f7961b3c5a372d4e1b269f35c29515f17